extensions/scarb-doc/src/lib.rs

```diff
@@ -29,15 +34,35 @@ pub struct AdditionalMetadata {
 pub fn generate_packages_information(
     metadata: &Metadata,
     metadata_for_packages: &[PackageMetadata],
-) -> Vec<PackageInformation> {
+    enabled_features: FeaturesOpts,
+) -> Result<Vec<PackageInformation>, anyhow::Error> {
     let mut packages_information = vec![];
     for package_metadata in metadata_for_packages {
         let authors = package_metadata.manifest_metadata.authors.clone();
 
+        let features_manifest: BTreeMap<FeatureName, Vec<FeatureName>> =
```

Comment:
doing this may lead to potentially incomplete information, because who knows when Scarb decides to add some custom logic to features collection/generation

it would be much better to collect all features via `scarb metadata`. if necessary fields are missing, the feel free to add them there (but please do it in separate PR)

Reply:
Your right, i looked through the code and will create separate PR before merging this one, that will collect features manifest for each package in metadata command

Reply:
Consult any changes to `scarb metadata` with @maciektr

Reply:
After discussion with @maciektr, we decided to use the metadata command directly, and make the behaviour of the missing features inside the `Scarb.toml` for separate task